Compile failes without enable-hunspell: spellchecker.cpp:111:80: error: cannot convert ‘std::string’ {aka ‘std::__cxx11::basic_string<char>’} to ‘const char*’

`cmake -C /var/tmp/portage_zram/portage/kde-misc/maliit-keyboard-2.1.0/work/keyboard-2.1.0_build/gentoo_common_config.cmake -G Ninja -DCMAKE_INSTALL_PREFIX=/usr -DKDE_INSTALL_USE_QT_SYS_PATHS=ON -DKDE_INSTALL_DOCBUNDLEDIR=/usr/share/help -Denable-tests=OFF -Denable-presage=OFF -Denable-hunspell=no -DCMAKE_DISABLE_FIND_PACKAGE_Qt5Feedback=ON -DCMAKE_DISABLE_FIND_PACKAGE_Anthy=ON -DCMAKE_DISABLE_FIND_PACKAGE_Pinyin=ON -DCMAKE_DISABLE_FIND_PACKAGE_Chewing=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_TOOLCHAIN_FILE=/var/tmp/portage_zram/portage/kde-misc/maliit-keyboard-2.1.0/work/keyboard-2.1.0_build/gentoo_toolchain.cmake  /var/tmp/portage_zram/portage/kde-misc/maliit-keyboard-2.1.0/work/keyboard-2.1.0`

```
/var/tmp/portage_zram/portage/kde-misc/maliit-keyboard-2.1.0/work/keyboard-2.1.0/plugins/westernsupport/spellchecker.cpp: In member function ‘void SpellCheckerPrivate::addUserDictionary(const QString&)’:
/var/tmp/portage_zram/portage/kde-misc/maliit-keyboard-2.1.0/work/keyboard-2.1.0/plugins/westernsupport/spellchecker.cpp:111:80: error: cannot convert ‘std::string’ {aka ‘std::__cxx11::basic_string<char>’} to ‘const char*’
  111 |                 hunspell->add(codec->fromUnicode(stream.readLine()).toStdString());
      |                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
      |                                                                                |
      |                                                                                std::string {aka std::__cxx11::basic_string<char>}
/var/tmp/portage_zram/portage/kde-misc/maliit-keyboard-2.1.0/work/keyboard-2.1.0/plugins/westernsupport/spellchecker.cpp:47:13: note:   initializing argument 1 of ‘int Hunspell::add(const char*)’
   47 |     int add(const char *) { return 0; }
```

--- a/plugins/westernsupport/spellchecker.cpp
+++ b/plugins/westernsupport/spellchecker.cpp
@@ -41,10 +41,9 @@ public:
     Hunspell(const char *, const char *, const char * = NULL) : encoding("UTF-8") {}
     int add_dic (const char *, const char * = NULL) { return 0; }
     char *get_dic_encoding() { return encoding.data(); }
-    int spell(const char *, int * = NULL, char ** = NULL) { return 1; }
-    int suggest(char *** lst, const char *) { if (lst) { *lst = NULL; } return 0; }
-    void free_list(char ***, int) {}
-    int add(const char *) { return 0; }
+    bool spell(const std::string&, int* = NULL, std::string* = NULL) { return true; }
+    std::vector<std::string> suggest(const std::string&) { return std::vector<std::string>(); }
+    int add(const std::string&) { return 0; }
 private:
     // Using QByteArray here instead of just returning "UTF-8" in get_dic_encoding
     // to avoid a following warning:
@@ -207,20 +206,18 @@ QStringList SpellChecker::suggest(const QString &word,
         return QStringList();
     }
 
-    const auto suggestions = d->hunspell->suggest(d->codec->fromUnicode(word).toStdString());
-
-    // Less than zero means some error.
-    if (suggestions.empty()) {
-        qWarning() << __PRETTY_FUNCTION__ << ": Failed to get suggestions for" << word << ".";
-        return QStringList();
-    }
+    auto suggestions = d->hunspell->suggest(
+                                d->codec->fromUnicode(word).toStdString());
 
     QStringList result;
-    const int final_limit((limit < 0) ? suggestions.size() : qMin(limit, static_cast<int>(suggestions.size())));
 
-    for (int index(0); index < final_limit; ++index) {
-        result << d->codec->toUnicode(suggestions[index].c_str());
+    for (auto const & s: suggestions) {
+        if (result.size() == limit)
+            break;
+
+        result.append(d->codec->toUnicode(s.data(), s.size()));
     }
+
     return result;
 }
 

